src/main/java/org/apache/sling/resourceresolver/impl/mapping/MapEntries.java

```diff
@@ -189,10 +199,16 @@ protected void doInit() {
 
             final Map<String, List<MapEntry>> newResolveMapsMap = new ConcurrentHashMap<>();
 
+            isAliasMapInitialized = false;
             //optimization made in SLING-2521
             if (this.factory.isOptimizeAliasResolutionEnabled()) {
-                final Map<String, Map<String, String>> aliasMap = this.loadAliases(resolver);
-                this.aliasMap = aliasMap;
+            	aliasTraversal = new Thread(new Runnable(){
+    				public void run() {
+    					aliasMap = loadAliases(resolver);
+    					isAliasMapInitialized = true;
+    					}
+    				});
+            	aliasTraversal.start();
```

Comment:
Are we missing an `else` block here which loads the aliases in the main thread?

Reply:
@rombert  - the patch only affects the if block - in the other case none of the query based caching happens and nothing needs to be loaded.

Reply:
Ah right. Thanks!